alacritty/src/renderer/text/builtin_font.rs

```diff
@@ -500,106 +506,50 @@ fn box_drawing(character: char, metrics: &Metrics, offset: &Delta<i8>) -> Raster
-    // g(x) = H - x + 1 lines. The intersection happens when f(x) = g(x), which is at
-    // x = H/2 (`y_center`).
-    let from_y = 1;
-    let x_end = y_center.floor();
-    let y_end = (height - from_y - 1) as f32;
-
-    // Pick the start point outside of the canvas to even-out the start.
-    let from_x = 0.;
-    let to_x = x_end;
-    canvas.draw_line_grid(from_x, from_y as f32, to_x, y_center.floor());
-    canvas.draw_line_grid(from_x, y_end, to_x, y_center.ceil());
-
-    // For regular arrows we handle thickness by drawing 2 angle arrows and then just filling
-    // the contents between them.
-    if (character == '\u{e0b1}' || character == '\u{e0b3}') && stroke_size > 1. {
-        // The default line is of stroke size 1, so the 0.5 is computed by subtracting 1 from
-        // stroke_size and then adding 0.5 to to put the target in the center of the cell.
-        let to_x = x_end - stroke_size;
-        canvas.draw_line_grid(from_x, from_y as f32 + stroke_size, to_x, y_center.floor());
-        canvas.draw_line_grid(from_x, y_end - stroke_size, to_x, y_center.ceil());
-    }
+    let slope = 1;
+    let f_y0 = 1;
+    let g_y0 = height as i32 - f_y0 - 1;
 
-    let buffer = canvas.buffer_mut();
-    if character == '\u{e0b0}' || character == '\u{e0b2}' {
-        for row in from_y..height - from_y {
-            let row_offset = row * width;
-            for index in 1..width {
-                let index = row_offset + index;
-                if buffer[index - 1]._r > buffer[index]._r && buffer[index]._r == 0 {
-                    break;
-                }
-
-                buffer[index - 1] = COLOR_FILL;
-            }
-        }
-    } else if stroke_size > 1. {
-        // Find the bottom/top most points of extra line we draw, so we can properly set the
-        // `start`.
-
-        let mut y1 = 0;
-        for row in (0..height / 2).rev() {
-            if buffer[row * width]._r != 0 {
-                y1 = row;
-                break;
-            }
-        }
-        let mut y2 = height / 2;
-        for row in height / 2..height {
-            if buffer[row * width]._r != 0 {
-                y2 = row;
+    // Start with offset `1` and draw until the intersection of the f(x) = x + 1 and
+    // g(x) = H - x - 1 lines. The intersection happens when f(x) = g(x), which is at
+    // x = H/2 - 1.
+    let intersection = (height as i32 + 1) / 2 - 1;
+
+    let f_x = LineEquation::new(slope, f_y0, 0, intersection);
+    let g_x = LineEquation::new(-slope, g_y0, 0, intersection);
+
+    // Inner functions to make arrows thicker.
+    let mut f_x_inner =
+        LineEquation::new(slope, f_y0 + extra_thickness, 0, intersection - extra_thickness);
+    let mut g_x_inner =
+        LineEquation::new(-slope, g_y0 - extra_thickness, 0, intersection - extra_thickness);
+
+    // NOTE f_x and g_x have the same amount of iterations.
+    for (p1, p2) in iter::zip(f_x, g_x) {
```

Comment:
```suggestion
    for (p1, p2) in f_x.zip(g_x) {
```

I think this should just work since `LineEquation` implements iterator and it's shorter?